editoast/src/schema/operation/mod.rs

```diff
@@ -28,31 +29,31 @@ pub enum Operation {
     Delete(DeleteOperation),
 }
 
-#[derive(Clone, Serialize)]
-#[serde(tag = "operation_type")]
-pub enum OperationResult {
-    #[serde(rename = "CREATE")]
-    Create(RailjsonObject),
-    #[serde(rename = "UPDATE")]
-    Update(RailjsonObject),
-    #[serde(rename = "DELETE")]
+#[derive(Clone)]
+pub enum CacheOperation {
+    Create(ObjectCache),
+    Update(ObjectCache),
     Delete(ObjectRef),
 }
 
 impl Operation {
-    pub async fn apply(&self, infra_id: i64, conn: &mut PgConnection) -> Result<OperationResult> {
+    pub async fn apply(
+        &self,
+        infra_id: i64,
+        conn: &mut PgConnection,
+    ) -> Result<Option<RailjsonObject>> {
         match self {
             Operation::Delete(deletion) => {
                 deletion.apply(infra_id, conn).await?;
-                Ok(OperationResult::Delete(deletion.clone().into()))
+                Ok(None)
             }
             Operation::Create(railjson_object) => {
                 create::apply_create_operation(railjson_object, infra_id, conn).await?;
-                Ok(OperationResult::Create(*railjson_object.clone()))
+                Ok(Some(railjson_object.deref().clone()))
             }
             Operation::Update(update) => {
                 let obj_railjson = update.apply(infra_id, conn).await?;
-                Ok(OperationResult::Update(obj_railjson))
+                Ok(Some(obj_railjson))
```

Comment:
minor for consistency with create above:
```suggestion
                Ok(Some(railjson_object))
```

Reply:
Done in https://github.com/osrd-project/osrd/compare/974d16044d52b0aa6b5d4ca92165ae23f570e057..551a69ba6f2557ae3511c13079ce670119a521ef. Thanks for catching that.